appinventor/components/src/com/google/appinventor/components/runtime/TextBoxBase.java

```diff
@@ -104,14 +112,21 @@ public TextBoxBase(ComponentContainer container, EditText textview) {
     // only). Maybe we need another color value which would be 'SYSTEM_DEFAULT' which
     // will not attempt to explicitly initialize with any of the properties with any
     // particular value.
-    // BackgroundColor(Component.COLOR_NONE);
     Enabled(true);
     fontTypeface = Component.TYPEFACE_DEFAULT;
     TextViewUtil.setFontTypeface(view, fontTypeface, bold, italic);
     FontSize(Component.FONT_DEFAULT_SIZE);
     Hint("");
+    if (isHighContrast || container.$form().HighContrast()) {
+      view.setHintTextColor(COLOR_YELLOW);
+    }
+    else {
+      view.setHintTextColor(0xFF9E9E9E);
```

Comment:
Where did 0xFF9E9E9E come from? Should it perhaps be defined in Component.java as a COLOR?

Reply:
Actually I was meaning to ask! Do you know the hex code for the default Hint text color? I couldn't find it in App Inventor documentation. Once I find it I'll add it to Component.java as suggested.

Reply:
It looks like before your change we didn't touch the hint text color. So, you can just not set it if not in high contrast mode (in the constructor). However, you will need to know what it is so that setHighContrast() will know what to do if it is called with false as the argument. To solve this, in the constructor you can call view.getHintTextColor().getDefaultColor() to learn what it is and save that in a variable (instance variable).